### PR TITLE
Add missing documentation for std::char types

### PR DESCRIPTION
**Pull request by GuillaumeGomez**

### CODE REVIEW

File: library/core/src/char/convert.rs [hunk 1]
```diff
@@ -193,6 +193,9 @@ impl From<u8> for char {
 }
 
 /// An error which can be returned when parsing a char.
+///
+/// This `struct` is created when using the
+/// [`char::from_str`](../../std/primitive.char.html#method.from_str) method.
```
Comment (jyn514) [inline library/core/src/char/convert.rs:198]: Please use intra-doc links so this doesn't break on re-exports.
```suggestion
/// [`char::from_str`] method.
```

File: library/core/src/char/convert.rs [hunk 2]
```diff
@@ -257,7 +260,11 @@ impl TryFrom<u32> for char {
     }
 }
 
-/// The error type returned when a conversion from u32 to char fails.
+/// The error type returned when a conversion from [`prim@u32`] to [`prim@char`] fails.
+///
+/// This `struct` is created by the
+/// [`char::try_from<u32>`](../../std/primitive.char.html#impl-TryFrom<u32>) method. See its
```
Comment (jyn514) [inline library/core/src/char/convert.rs:266]:
```suggestion
/// [`char::try_from<u32>`](char#impl-TryFrom<u32>) method. See its
```

File: library/core/src/char/convert.rs (hunk 1 above)
Comment (GuillaumeGomez) [reply library/core/src/char/convert.rs:198]: I don't even know why I didn't use it directly...

**Comment by GuillaumeGomez** [comment]
I forgot that we don't support for primitive types yet... Needs to wait for https://github.com/rust-lang/rust/pull/87073 to be merged.

**Comment by GuillaumeGomez** [comment]
@jyn514: ^ wasn't this restriction lifted?

**Comment by jyn514** [comment]
@GuillaumeGomez it needs to wait for the bootstrap compiler to be bumped, so another 6 weeks. To avoid waiting longer (e.g. if it turns out rustdoc is buggy), you could run `x.py doc --stage 1 library/std` to make sure it will work at that time.

**Comment by GuillaumeGomez** [comment]
I did and it worked, which is why I was confused. Well, let's wait for 6 weeks then!

**Comment by jyn514** [comment]
@GuillaumeGomez it's been 6 weeks :)

**Comment by GuillaumeGomez** [comment]
Let's see! :D

File: library/core/src/char/convert.rs [hunk 3]
```diff
@@ -199,6 +199,8 @@ impl const From<u8> for char {
 }
 
 /// An error which can be returned when parsing a char.
+///
+/// This `struct` is created when using the [`char::from_str`](char#tymethod.from_str) method.
```
Comment (GuillaumeGomez) [inline library/core/src/char/convert.rs:203]: I think it's still not ready:

```
warning: unresolved link to `char::from_str`
 --> foo.rs:1:7
  |
1 | /// [`char::from_str`]
  |       ^^^^^^^^^^^^^^ the builtin type `char` has no associated item named `from_str`
```

We need https://github.com/rust-lang/rust/issues/90703 to be fixed first.

Comment (GuillaumeGomez) [inline library/core/src/char/convert.rs:203]: I think it's still not ready:

```
warning: unresolved link to `char::from_str`
 --> foo.rs:1:7
  |
1 | /// [`char::from_str`]
  |       ^^^^^^^^^^^^^^ the builtin type `char` has no associated item named `from_str`
```

We need https://github.com/rust-lang/rust/issues/90703 to be fixed first.

Comment (jyn514) [reply library/core/src/char/convert.rs:203]: Wait, this doesn't match the error above though. The CI failure resolved the link correctly, it's just a dead link for some reason. What version of rustdoc did you use to test locally? Does the error change if you use beta?

Comment (jyn514) [reply library/core/src/char/convert.rs:203]: Oh, it's because you used a fragment link instead of an intra-doc link.

Comment (GuillaumeGomez) [reply library/core/src/char/convert.rs:203]: Yes, which is invalid too. I'll update the current code but even the fixed version is broken for the moment.

**Comment by camelid** [comment]
> We need #90703 to be fixed first.

#90703 is fixed now. I think this PR is no longer blocked?

**Comment by GuillaumeGomez** [comment]
Let's give it another try!

**Comment by GuillaumeGomez** [comment]
Seems like the CI passed this time. \o/

**Comment by GuillaumeGomez** [comment]
Forgot to r+ it...

@bors: r=jyn514,camelid

**Pull request merged**